hibernate-core/src/test/java/org/hibernate/orm/test/orphan/onetomany/DeleteOneToManyOrphansTest.java

```diff
@@ -155,4 +156,17 @@ public void testReplacedWhileManaged(SessionFactoryScope scope) {
 		);
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HHH-15258")
+	public void testPersistAndQuery(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					Product product = new Product();
+					session.persist( product );
+					assertNotNull( session.createQuery( "from Product" ).list() );
+					assertNotNull( session.createQuery( "from Product" ).list() );
```

Comment:
If I didn't misunderstand, your concern is the second `list()` statement would throw exception. It would be great to convey your assertion logic more cleanly, e.g.:
```
session.persist( product );
session.createQuery( "from Product" ).list();
assertDoesNotThrow(session.createQuery( "from Product" ).list(), "without fixing, an exception would be thrown for the second list() invocation");
```
or simply:
```
session.persist( product );
session.createQuery( "from Product" ).list();
// without fixing, the following statement would throw exception
session.createQuery( "from Product" ).list();
```
It would save unnecessary effort for future code maintainer to figure out the irrelevant `assertNotNull()` logic. IMHO.